environment/stats/src/cloud/benchflow/collectors/stats.go

```diff
@@ -20,11 +20,12 @@ import (
 type Container struct {
 	ID           string
 	statsChannel chan *docker.Stats
+	doneChannel chan bool
```

Comment:
Why do you need again a per container based `doneChannel`? When we stop the collector, this holds for all the containers.